src/smartpeak/source/core/RawDataProcessors/FitFeaturesEMG.cpp

```diff
@@ -130,29 +130,33 @@ namespace SmartPeak
         }
         OpenMS::ConvexHull2D hull;
         hull.addPoints(hull_points);
-        subfeature.getConvexHulls().push_back(hull);
+        // Bug in OpenMS Feature::getConvexHulls() returns the bounding box
+        // when more than a single convex hull is detected, which does not allow for plotting.
+        //subfeature.getConvexHulls().push_back(hull); 
+        std::vector<OpenMS::ConvexHull2D> hulls{ hull };
+        subfeature.setConvexHulls(hulls);
 
         OpenMS::PeakIntegrator pi;
-        std::cout << "Updating ranges...\n";
+        LOGD << "Updating ranges...";
         emg_chrom.updateRanges();
-        std::cout << "Ranges updated.\n";
+        LOGD << "Ranges updated.";
         const double emg_chrom_left { emg_chrom.getMinRT() };
         const double emg_chrom_right { emg_chrom.getMaxRT() };
-        std::cout << "Positions calculated.\n";
+        LOGD << "Positions calculated.";
         OpenMS::PeakIntegrator::PeakArea pa = pi.integratePeak(emg_chrom, emg_chrom_left, emg_chrom_right);
-        std::cout << "Area calculated.\n";
+        LOGD << "Area calculated.";
         OpenMS::PeakIntegrator::PeakBackground pb = pi.estimateBackground(emg_chrom, emg_chrom_left, emg_chrom_right, pa.apex_pos);
-        std::cout << "Background calculated.\n";
+        LOGD << "Background calculated.";
         double peak_integral { pa.area - pb.area };
         double peak_apex_int { pa.height - pb.height };
         if (peak_integral < 0) { peak_integral = 0; }
         if (peak_apex_int < 0) { peak_apex_int = 0; }
 
-        std::cout << "Intensity: " << subfeature.getIntensity() << "\t" << peak_integral << "\n";
-        std::cout << "peak_apex_position: " << subfeature.getMetaValue("peak_apex_position") << "\t" << pa.apex_pos << "\n";
-        std::cout << "peak_apex_int: " << subfeature.getMetaValue("peak_apex_int") << "\t" << peak_apex_int << "\n";
-        std::cout << "area_background_level: " << subfeature.getMetaValue("area_background_level") << "\t" << pb.area << "\n";
-        std::cout << "noise_background_level: " << subfeature.getMetaValue("noise_background_level") << "\t" << pb.height << "\n\n";
+        LOGD << "Intensity: " << subfeature.getIntensity() << "\t" << peak_integral;
+        LOGD << "peak_apex_position: " << subfeature.getMetaValue("peak_apex_position") << "\t" << pa.apex_pos;
```

Comment:
Looks like there is an ambiguity at least on linux for the 4 following line. Perhaps you can try to cast to either int, short int or bool

Reply:
I see.  OK I should have looked a bit deeper into the logs.  I will take care of this in a few hours I think on the next push.

Reply:
@bertrandboudaud I fixed the build issues on MacOS and Linux.  However, Windows is failing, but the build output is more than 50 MB so I am not sure what the issue is.  Builds on my machine :).  Have you used the `store_artifacts` to save the build log or is there a simpler way to find out what the issue is?

Reply:
@bertrandboudaud I figured out what the issue was.